packages/sidenav/src/ux-sidenav/ux-sidenav.ts

```diff
@@ -0,0 +1,107 @@
+import { customElement, useView, inject, TaskQueue, PLATFORM, bindable } from "aurelia-framework";
+import { UxSidenavDrawer, OPEN_CHANGED } from "../ux-sidenav-drawer/ux-sidenav-drawer";
+import { UxSidenavTheme } from "./ux-sidenav-theme";
+import { StyleEngine, normalizeNumberAttribute } from "@aurelia-ux/core";
+import { IUxSidenavDrawerElement } from "../ux-sidenav-drawer/i-ux-sidenav-drawer-element";
+import { UxDefaultSidenavConfiguration } from "../ux-default-sidenav-configuration";
+
+@inject(Element, StyleEngine, TaskQueue, UxDefaultSidenavConfiguration)
+@customElement('ux-sidenav')
+@useView(PLATFORM.moduleName('./ux-sidenav.html'))
+export class UxSidenav {
+  constructor(public element: HTMLElement, private styleEngine: StyleEngine, private taskQueue: TaskQueue,
+    private defaultConfig: UxDefaultSidenavConfiguration) { }
+
+  backdrop: boolean = this.defaultConfig.backdrop;
+  width: number;
+
+  leftDrawer: UxSidenavDrawer | undefined;
+  rightDrawer: UxSidenavDrawer | undefined;
+  bottomDrawer: UxSidenavDrawer | undefined;
```

Comment:
Another edge case that I can imagine... 

A situation where the consumer needs several `left` (or any other position) drawers. From what I understand in `ux-sidenav.ts` it would be possible to achieve this with having an array of `drawers` instead of 3 distincts properties for `leftDrawer`,  `rightDrawer` and `bottomDrawer`?

Reply:
ah, this... no, not supported :)

Reply:
can enhance later... i'd like to merge all three branches first, it's getting harder to juggle them

Reply:
I'm ok with this.

Reply:
I don't insist, happy to change it